### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by DangerOnTheRanger**
This PR adds some basic end-user documentation. It mentions features from #8, which should be merged first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,75 @@
 celvet
 ======
 
+**celvet** is a multitool for linting CRDs (`CustomResourceDefinition`) in
```
Comment (jpbetz) [inline README.md:4]: What does multitool mean?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,75 @@
...
+checking that they do not exceed the resource constraints provided by
+Kubernetes' API server. The tool is deliberately made to be useful both in an
+ad-hoc fashion by developers in addition to being part of a CI/CD pipeline, so
+`celvet` supports output in JSON in addition to machine-readable text.
+
+What does celvet catch?
+-----------------------
+
+* **CRD parsing errors** - celvet will early exit if it finds that a CRD
+    contains invalid YAML/JSON.
```
Comment (jpbetz) [inline README.md:15]: There's a big difference between validating that YAML/JSON is valid and that the YAML/JSON is a valid CRD.  Does this do the later?

File: README.md [hunk 3]
```diff
@@ -1,12 +1,75 @@
...
+* **Missing limits on variable-length fields** - `maxLength`, `maxItems`, etc.
+* **Validation rule parsing errors**  - if celvet has issues parsing a
+    validation rule, it will output the errors.
```
Comment (jpbetz) [inline README.md:18]: Clarify what a validation rule is (either via a link or stating that this is about x-kubernetes-validations, or both)?

File: README.md [hunk 4]
```diff
@@ -1,12 +1,75 @@
...
+* **Exceeding resource constraints for validation rules** - if a validation
+    rule exceeds the per-rule cost budget, celvet will output an error.
```
Comment (jpbetz) [inline README.md:20]: Similar to above, clarify what a per-rule cost budget is?

File: README.md [hunk 5]
```diff
@@ -1,12 +1,75 @@
...
+The `-o`/`--output` option can be used to change the output format to JSON if
+`json` is passed to the flag; `text` is the default option but can be
+explicitly passed. The same file but with `-o json` passed will give the
+following (the data has been pretty-printed here for readability):
+```json
+{
+  "limitErrors": [
+    {
+      "path": "<root>.spec.cronSpec",
+      "type": 2
```
Comment (jpbetz) [inline README.md:55]: What are users expected to do with the type numbers?

File: README.md [hunk 6]
```diff
@@ -1,12 +1,75 @@
...
+    },
+    {
+      "path": "<root>.spec.image",
+      "type": 2
+    }
+  ],
+  "costErrors": null,
+  "compileErrors": null,
+  "otherErrors": []
```
Comment (jpbetz) [inline README.md:64]: Why are the 1st two null and the last an empty array?

File: README.md [hunk 7]
```diff
@@ -1,12 +1,75 @@
...
+```
+
+The full output of `celvet -h` is given below.
+
+```
+./celvet [flags] crd-file
+  -r, --human-readable   print out values in human-readable formats (only applies if --json/-j is not passed) (default true)
+  -o, --output string    output format (valid values are "text" and "json") (default "text")
+pflag: help requested
+```
```
Comment (jpbetz) [inline README.md:75]: I'm concerned this part of the readme is going to get stale and become more misleading than it is helpful.

**Review by jpbetz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (DangerOnTheRanger) [reply README.md:4]: [This](https://en.wikipedia.org/wiki/Multi-tool) is what I was going for (to get the idea across that celvet can/will be able to solve a lot of different use cases), but maybe a different introductory sentence would be better.

File: README.md (hunk 6 above)
Comment (DangerOnTheRanger) [reply README.md]: This was using an old commit that didn't include https://github.com/DangerOnTheRanger/celvet/pull/8/commits/2cf6fd9e410572d7d4c2c0ffa15bd0162a03bf9e - I've re-linted the same file from that branch, so the output should be cleaner now.

File: README.md (hunk 5 above)
Comment (DangerOnTheRanger) [reply README.md]: I've re-linted the file with a binary that includes https://github.com/DangerOnTheRanger/celvet/pull/8/commits/2fc19fc5efabdfb8d481cdedd3745a3b609a17d1 - fixed

File: README.md (hunk 7 above)
Comment (DangerOnTheRanger) [reply README.md]: Gotcha, I've removed that part entirely.

File: README.md (hunk 3 above)
Comment (DangerOnTheRanger) [reply README.md]: Fixed - I've added a link to the k8s documentation on validation rules.

File: README.md (hunk 4 above)
Comment (DangerOnTheRanger) [reply README.md]: Gotcha, I've added a link to the relevant documentation here as well.

File: README.md (hunk 2 above)
Comment (DangerOnTheRanger) [reply README.md:15]: Not at the moment - validation rules are linted, and the YAML/JSON itself is parsed of course, but that's not enough to claim that CRDs themselves are validated in my opinion.

Comment (jpbetz) [reply README.md:15]: Okay, that's what I understood as well. The phrase "CRD parsing errors" doesn't feel entirely accurate here.

Are you tracking the lack of CRD validation support as an issue?

File: README.md [hunk 8]
```diff
@@ -1,12 +1,63 @@
...
+string "spec.validation.openAPIV3Schema.properties[spec].properties[image]" missing maxLength
+```
+
+The `-o`/`--output` option can be used to change the output format to JSON if
+`json` is passed to the flag; `text` is the default option but can be
+explicitly passed. The same file but with `-o json` passed will give the
+following (the data has been pretty-printed here for readability):
+```json
+{
+  "limitErrors": [
```
Comment (jpbetz) [inline README.md:52]: Apologies for being this up so late in the review process, but do we really want a different group for each type of error? As more and more linter rules are added, will this result a large number of different error groups that a client must know how to process?

Comment (jpbetz) [reply README.md:52]: Also, how does a user know what "limitErrors" are and how to fix them? (same question applies to all error categories)

**Comment by jpbetz** [comment]
Feel free once issue is opened for https://github.com/DangerOnTheRanger/celvet/pull/9#discussion_r956493337 and the documentation is updated to directly explain what `limitErrors` (and all the other categories) contain and how to fix them (use a table maybe?)

File: README.md (hunk 2 above)
Comment (DangerOnTheRanger) [reply README.md:15]: There's brief mention of it in the (admittedly non-public at the moment) design document, but I've added #11 to track it and reworded the line to better indicate that just the syntax of the YAML/JSON is validated as opposed to the CRD proper.

File: README.md (hunk 8 above)
Comment (DangerOnTheRanger) [reply README.md:52]: I've added a list to the README of the different linter error types and opened #10 to track fixing the JSON output - fixed

**Pull request merged**